### DISCUSSION TITLE
Aggregated trajectories - flow direction

### DISCUSSION THREAD

**Discussion by Mrvy29** [anitagraser/movingpandas#273] [Q&A] [answered]
Hi all, 
love this package a lot!

I am aggregating trajectories and would really like to see the flows direction as well, is this implemented ?
Im assuming the flows cover all trajectories starting or ending at a cluster point, and they are bundled up, is there a way to differentiate between the two ?

Thanks!

**Comment by raybellwaves** [comment]
Do you have some dummy trajectories data you can share and what the actual expected flow direction would be?

**Comment by anitagraser** [comment]
The flow direction is considered in the aggregations and results. However, the challenge is to visualize the flow direction in the map plots since geoviews has no line offset functionality

**Comment by Mrvy29** [comment]
Thank you,
I understand.
Do you think it would be possible to visualize in seperate plots?
How is that information extracted from the aggregator?



[image: Seafloor Scene Icon]

**Reply by anitagraser** [reply, marked as answer]
> Do you think it would be possible to visualize in seperate plots? 

The problem is tricky. What logic could be used to split the data into multiple plots?

> How is that information extracted from the aggregator? 

There is a convenience function called get_flow_gdf() that returns a GeoDataFrame with Linestrings:

https://github.com/anitagraser/movingpandas-examples/blob/main/3-tech-demos/scikit-learn-od-clustering.ipynb

![image](https://user-images.githubusercontent.com/590385/207139859-89cd82c1-ac7c-442e-a894-42db99582826.png)

![image](https://user-images.githubusercontent.com/590385/207139990-dba52edb-fc13-44df-bf72-21e05082a70d.png)

**Reply by Mrvy29** [reply]
Thank you so much for taking the time to answer, 
I have looked through the linestrings and what I am struggling to understand is how exactly the lines represent the direction.
I'm assuming each line represents the flow from cluster point n1 to cluster point n2 for example, does it mean there would possibly be a separate line that represents flow from cluster point n2 to n1 ? and basically the difference is the weight and order of the points in linestrings?

**Reply by anitagraser** [reply]
Yes, if there are flows from n1->n2 as well as from n2->n1, then there will be two lines with different order of points and independent weights
